src/sentry/ingest/billing_metrics_consumer.py

```diff
@@ -141,5 +134,31 @@ def _produce_billing_outcome(
             quantity=quantity,
         )
 
+    def _flag_metric_received_for_project(self, generic_metric: GenericMetric) -> None:
+        project_id = generic_metric["project_id"]
+
+        # We try to extract the MRI from the metric_id since our goal is to check whether the MRI belongs to a metric
+        # in the `custom` namespace.
+        metric_mri = self._resolve(generic_metric["mapping_meta"], generic_metric["metric_id"])
+        parsed_mri = parse_mri(metric_mri)
+        if parsed_mri is None or not is_custom_metric(parsed_mri):
+            return
```

Comment:
Should you do this *after* checking if the project already has the flag set? If the flag *is* set you can skip all this work as we don't care about the results.

Reply:
Yeah, that is a good idea. Thanks for the suggestion!